utils/bigclown/bigclown-gateway/files/init

```diff
@@ -0,0 +1,57 @@
+#!/bin/sh /etc/rc.common
+
+START=98
+STOP=1
+
+USE_PROCD=1
+
+PROG=/usr/bin/bcg
+CONF=/tmp/etc/bigclown-gateway.conf
+
+append() {
+	local cfg="$1"
+	local uci_name="$2"
+	local out_name="$3"
+	local default="$4"
+	config_get val $cfg $uci_name $default
+	if [ -n "$val" ]; then
+	  echo "$out_name $val" >> $CONF
+	fi
+}
+
+start_service() {
+	config_load bigclown-gateway
+
+	rm -rf $CONF
+	echo "Generating bigclown-gateway config file in $CONF"
+
+	append gateway name 'name:' usb-dongle
+	# TODO add udev rules and use different default here
+	append gateway device 'device:' /dev/ttyUSB0
```

Comment:
but yes, the defaults seem..... rather particular.

Reply:
It would be better to have default that works only on bigclown usb dongle but I have to first write good hotplug script for that. I will see what I can do.